libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/ElectronicHorizonRequestBuilder.java

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.location.Location;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ElectronicHorizonRequestBuilder {
+  private Gson gson = new Gson();
+
+  public String build(Expansion expansion, List<Location> locations) {
+    List<Position> positions = new ArrayList<>();
+    for (Location location : locations) {
+      positions.add(new Position(location.getLatitude(), location.getLongitude()));
+    }
+
+    Map<String, Object> options = new HashMap<>();
+    options.put("expansion", expansion.value);
+
+    return gson.toJson(new ElectronicHorizonRequest(positions, options));
+  }
+
+  public enum Expansion {
+    _1D("1D"),
+    _1_5D("1.5D"),
+    _2D("2D");
+
+    private String value;
+
+    Expansion(String expansion) {
+      this.value = expansion;
+    }
+  }
+
+  static class Position {
```

Comment:
Does `Position` need to be `static`? Actually it can be `private`.

Reply:
fixed